imports/server/garbage-collection.ts

```diff
@@ -37,15 +37,18 @@ function cleanup() {
   }
 
   // Run all hooks.
-  globalGCHooks.forEach((f) => f(deadServers));
+  await globalGCHooks.reduce(async (p, f) => {
+    await p;
+    await f(deadServers);
+  }, Promise.resolve());
```

Comment:
Hat tip to [this StackOverflow answer](https://stackoverflow.com/a/24586168) (and to VSCode's "convert this function to async/await" helper) for this pattern, which I found to be quite pleasant for cases where you basically want to run through a loop and have each step run sequentially.